internal/transport/http2_client.go

```diff
@@ -1023,6 +1023,14 @@ func (t *http2Client) Close(err error) {
 	case <-timer.C:
 		t.logger.Infof("Failed to write a GOAWAY frame as part of connection close after %s. Giving up and closing the transport.", goAwayLoopyWriterTimeout)
 	}
+	// Wait for the reader channel to close
+	select {
+	case <-t.readerDone:
+	default:
+		if t.logger.V(logLevel) {
+			t.logger.Infof("The reader channel has already been closed: %v", err)
```

Comment:
That's not what a `default` case means in a `select` with a channel.

I think we can pretty safely just close `conn` and do a blocking `<-t.readerDone`.

Reply:
Yes, I did that when I did not know that the tests were failing when t.Close() was being called from inside `reader` function. I will remove it and refactor the code to make it work.